src/main/java/com/amplitude/api/IdentifyInterceptor.java

```diff
@@ -0,0 +1,193 @@
+    private final WorkerThread logThread;
+
+    private long identifyBatchIntervalMillis;
+
+    private AtomicBoolean transferScheduled = new AtomicBoolean(false);
+
+    private long lastIdentifyInterceptorId = -1;
+
+    private final AmplitudeClient client;
+
+    public  IdentifyInterceptor (
+            DatabaseHelper dbHelper,
+            WorkerThread logThread,
+            long identifyBatchIntervalMillis,
+            AmplitudeClient client
+    ) {
+        this.dbHelper = dbHelper;
+        this.logThread = logThread;
+        this.identifyBatchIntervalMillis = identifyBatchIntervalMillis;
+        lastIdentifyInterceptorId = dbHelper.getLastIdentifyInterceptorId();
+        this.client = client;
+    }
+
+    public JSONObject intercept(String eventType, JSONObject event) {
+        if (eventType.equals(Constants.IDENTIFY_EVENT)) {
+            if (isSetOnly(event)) {
+                // intercept and  save user properties
+                lastIdentifyInterceptorId = saveIdentifyProperties(event);
+                scheduleTransfer();
+                return null;
+            } else if(isClearAll(event)){
+                // clear existing and return event
+                dbHelper.removeEvents(lastIdentifyInterceptorId);
+                return event;
+            } else {
+                // Fetch and merge event
+                return fetchAndMergeToIdentifyEvent(event);
+            }
+        } else if (eventType.equals(Constants.GROUP_IDENTIFY_EVENT)) {
+            // no op
+            return event;
+        } else {
+            // fetch, merge and attach user properties
+            return fetchAndMergeToNormalEvent(event);
+        }
+    }
+
+    public void setIdentifyBatchIntervalMillis(long identifyBatchIntervalMillis) {
+        this.identifyBatchIntervalMillis = identifyBatchIntervalMillis;
+    }
+
+    private JSONObject fetchAndMergeToIdentifyEvent(JSONObject event) {
+        try {
+            List<JSONObject> identifys = dbHelper.getIdentifyInterceptors(lastIdentifyInterceptorId, -1);
+            if (identifys.isEmpty()) {
+                return event;
+            }
+            JSONObject identifyEventUserProperties = event.getJSONObject("user_properties");
+            JSONObject userProperties = identifyEventUserProperties.has(Constants.AMP_OP_SET) ?
+                    identifyEventUserProperties.getJSONObject(Constants.AMP_OP_SET) :
+                    new JSONObject();
+            for (int i = 0; i < identifys.size(); i++) {
+                JSONObject identify = identifys.get(i);
+                JSONObject setUserProperties = identify.getJSONObject("user_properties")
+                        .getJSONObject(Constants.AMP_OP_SET);
+                Iterator<?> keys = setUserProperties.keys();
+                while (keys.hasNext()) {
+                    String key = (String) keys.next();
+                    userProperties.put(key, setUserProperties.get(key));
+                }
+            }
```

Comment:
For this, should we change it to apply the event properties last in case of overlap?